src/cargo/util/flock.rs

```diff
@@ -395,8 +395,38 @@ mod sys {
 
     #[cfg(target_os = "solaris")]
     fn flock(file: &File, flag: libc::c_int) -> Result<()> {
-        // Solaris lacks flock(), so simply succeed with a no-op
-        Ok(())
+        // Solaris lacks flock(), so try to emulate using fcntl()
+        let mut flock = libc::flock {
+            l_type: 0,
+            l_whence: 0,
+            l_start: 0,
+            l_len: 0,
+            l_sysid: 0,
+            l_pid: 0,
+            l_pad: [0, 0, 0, 0],
+        };
+        flock.l_type = if flag & libc::LOCK_UN != 0 {
+            libc::F_RDLCK
```

Comment:
Shouldn't this be `F_UNLCK`?

Reply:
Oh my, I didn't catch that. @psumbera can you follow up with a PR to update this?

Reply:
I will fix it. Sorry I didn't catch that. Since I have tested it this doesn't cause the problem with use in cargo.

Reply:
https://github.com/rust-lang/cargo/pull/11474